cmake/Modules/FindDPCPP.cmake

```diff
@@ -77,8 +82,20 @@ function(add_sycl_to_target)
     ${ARGN}
   )
   target_compile_options(${SB_ADD_SYCL_TARGET} PUBLIC ${DPCPP_FLAGS})
+  if (DPCPP_SYCL_TARGET MATCHES "nvptx64-nvidia-cuda")
+    target_compile_options(${SB_ADD_SYCL_TARGET} PUBLIC -Xsycl-target-backend 
+                              PUBLIC --cuda-gpu-arch=sm_80 
```

Comment:
Basically, if you compile every NVIDIA target for `sm_80`, you will get a runtime error in case of the actual device architecture is less than `sm_80`. Here again, I suggest setting the defines by checking the `DPCPP_SYCL_ARCH`.